roles/nis-client/tasks/main.yml

```diff
@@ -70,3 +70,25 @@
   file:
     path: /usr/sbin/policy-rc.d
     state: absent
+
+#
+# ncsd
+#
+
+- name: nscd service use_nscd={{ nis_client_use_nscd }}
+  service: name=nscd state=stopped
+  failed_when: False
+  when: not nis_client_use_nscd
+  tags: nscd
+
+- name: nscd packages use_nscd={{ nis_client_use_nscd }}
```

Comment:
The name of the package is `nscd` in both distros, so you may be able to refactor this to just use the `package` module.